api/v1beta1/flowcollector_types.go

```diff
@@ -255,6 +257,26 @@ type FlowCollectorIPFIXReceiver struct {
 	Transport string `json:"transport,omitempty"`
 }
 
+type FlowCollectorS3 struct {
+	// Important: Run "make generate" to regenerate code after modifying this file
+
+	//+kubebuilder:default:=""
+	// address of the s3 server
+	Endpoint string `json:"endpoint"`
+
+	//+kubebuilder:default:=""
+	// username to connect to server
+	AccessKeyID string `json:"accessKeyId"`
+
+	//+kubebuilder:default:=""
+	// password to connect to server
+	SecretAccessKey string `json:"secretAccessKey"`
```

Comment:
It would be better to load a secret here but it's useless for now as FLP doesn't.
In result, the credentials would be in clear text in the FLP configmap.

Reply:
Well, I think we should read it from a secret/CM anyway: not because it makes it more hidden (you're right that it will be visible in FLP anyway - which is ok as long as FLP is only accessible for cluster admins) , but in a gitops perspective users would like to decouple a config such as FlowCollector CR, and secrets management, for instance by using stuff like Vault: so that the FlowCollector CR can be safely stored in config repos (e.g. in git) with compromising secrets, and secrets are generated/injected at runtime.

I have a commit for Kafka / SASL that I haven't pushed as a PR yet (I should do that) where there's a similar problem to solve: https://github.com/jotak/network-observability-operator/commit/17a2ae96074b05cfbab57c3a9b66624e086a864a - maybe I should push it and we can align our 2 PRs & reuse some structs / functions

Reply:
(edit: rebased my sasl branch, new one is https://github.com/jotak/network-observability-operator/commit/3a4f105afb19ce39ac43410ff22ddbacda2f232f )

Reply:
I was considering replacing these creds by a secret and mounting the secret inside FLP. 
Do you feel it's not necessary here ?

Reply:
yes it can be done that way but I guess then you need to update FLP to read from file?
That's also what I did for sasl: https://github.com/jotak/network-observability-operator/commit/3a4f105afb19ce39ac43410ff22ddbacda2f232f#diff-adf3c77f719dd339bbc5c11eeb3e846bff5024d8f79a4f4461fa914743cb9c50R294-R312

Reply:
yeah I'm fine updating FLP to add this capability if we are aligned